src/xdocs/config_design.xml

```diff
@@ -1508,39 +1508,43 @@ class Test {
         </source>
         <p>Example with default values:</p>
         <source>
-public class MyClass {
- private int myPrivateField1;
+class Example1 {
+  private int myPrivateField1;
 
- int field1;              // violation, must have a visibility modifier
+  int field1; // violation '.* must be private .*'
 
- protected String field2; // violation, protected visibility is not allowed
+  protected String field2; // violation '.* must be private .*'
 
- public int field3 = 42;  // violation, not static final, immutable,
-                          // nor matching the pattern
+  public int field3 = 42; // violation '.* must be private .*'
 
- public long serialVersionUID = 1L;
+  public long serialVersionUID = 1L;
 
- public static final int field4 = 42;
+  public static final int field4 = 42;
 
- public final int field5 = 42;          // violation, public immutable fields are not allowed
- public final java.lang.String notes;   // violation, public immutable fields are not allowed
+  public final int field5 = 42; // violation '.* must be private .*'
 
- public final Set&lt;String&gt; mySet1 = new HashSet&lt;&gt;();  // violation, HashSet is mutable
+  public final java.lang.String notes; // violation '.* must be private .*'
 
- public final ImmutableSet&lt;String&gt; mySet2;           // violation,
-                                                     // immutable type not in config
+  public final Set&lt;String&gt; mySet1 = new HashSet&lt;&gt;(); // violation '.* must be private .*'
 
- public final ImmutableMap&lt;String, Object&gt; objects1; // violation,
-                                                     // immutable type not in config
+  public final ImmutableSet&lt;String&gt; mySet2; // violation '.* must be private .*'
 
- @com.annotation.CustomAnnotation
- String annotatedString;      // violation, annotation is not configured
+  public final ImmutableMap&lt;String, Object&gt; objects1; // violation '.* must be private .*'
 
- @CustomAnnotation
- String shortCustomAnnotated; // violation, annotation is not configured
+  @com.puppycrawl.tools.checkstyle.XdocsPropertyType(PropertyType.STRING)
+  String annotatedString; // violation '.* must be private .*'
 
- @com.google.common.annotations.VisibleForTesting
- public String testString = "";
+  @XdocsPropertyType(PropertyType.STRING)
+  String shortCustomAnnotated; // violation '.* must be private .*'
+
+  @com.google.common.annotations.VisibleForTesting
+  public String testString = "";
+
+  Example1(String notes, ImmutableSet&lt;String&gt; mySet2, ImmutableMap&lt;String, Object&gt; objects1) {
```

Comment:
I think I should omit the constructor from the xdocs, right?

Reply:
Is it because of final fields ?

Reply:
Yes it is.
Better to not avoid anything in class, it is too complicated.

Can you initialize fields with null ? In declaration.
To avoid ctor creation.

Reply:
Done. Removed it.